x-pack/plugins/observability/server/lib/rules/threshold/lib/metric_query.ts

```diff
@@ -25,7 +26,13 @@ const getParsedFilterQuery: (filterQuery: string | undefined) => Array<Record<st
   filterQuery
 ) => {
   if (!filterQuery) return [];
-  return [JSON.parse(filterQuery)];
+
+  try {
```

Comment:
@maryam-saeidi, this is where compile the `filterQuery` on the server side instead of `convertKueryToElasticSearchQuery` in the frontend, right?

Reply:
Right